internal/services/containers/kubernetes_flux_configuration_resource_test.go

```diff
@@ -231,6 +231,41 @@ func TestAccKubernetesFluxConfiguration_kustomizationPostBuildUpdate(t *testing.
 	})
 }
 
+func TestAccKubernetesFluxConfiguration_gitRepositoryProvider(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_kubernetes_flux_configuration", "test")
+	r := KubernetesFluxConfigurationResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.gitRepositoryProvider(data, string(fluxconfiguration.ProviderTypeAzure)),
```

Comment:
We should not be passing test variables to the configuration files as parameters due to code readability. Could we clone the configuration and hardcode the value in that configuration function? 
```suggestion
			Config: r.gitRepositoryProviderTypeAzure(data),
```

Reply:
Hey @WodansSon, was there a PR review where we said not to do this? I was about to leave a comment that this was fine but I just want to make sure it wasn't said elsewhere